cypress-io/cypress
# feat: Add login state to manifest request for analytics

- Closes #22754 

### User facing changelog
N/A

### Additional details
* Pass new `x-logged-in` header on check for updates manifest request for Product analytics

### Steps to test
Unfortunately, request headers aren't logged in our debug calls so there isn't an easy way to validate this as-is. As a workaround you can add `console.log(manifestHeaders)` somewhere [around here](https://github.com/cypress-io/cypress/compare/22754-add-login-state-to-manifest-request?expand=1#diff-f4439168518f09f406948b6f67428518c25a6ac04a9b766f86b2dfbe1964ac2eR165) in `VersionsDataSource` and follow this procedure:

1. Open the app, launchpad should appear
2. Check whether you are logged in within the app
3. Check your console log, the manifest headers should have output and you can confirm the value of the `x-logged-in` header is accurate for your login state
4. If logged in, logout or vice versa and repeat to confirm the header reflects correctly for both states

### How has the user experience changed?
No visible change

### PR Tasks

- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
